ext/SpeedyWeatherCUDAExt/fourier.jl

```diff
@@ -0,0 +1,119 @@
+    rfft_plans_1D::Vector{AbstractFFTs.Plan},
+    brfft_plans_1D::Vector{AbstractFFTs.Plan},
+    fake_grid_data::AbstractGridArray{NF, N, <:CuArray{NF}},
+    scratch_memory_north::CuArray{Complex{NF}},
+    rings::AbstractArray,
+    nlons::Vector{<:Int}
+) where {NF<:AbstractFloat, N}
+    # Determine which FFT package to use (currently either FFTW or GenericFFT)
+    FFT_package = SpeedyTransforms.which_FFT_package(CuArray{NF})
+
+    # For each ring generate an FFT plan (for all layers and for a single layer)
+    for (j, nlon) in enumerate(nlons)
+        real_matrix_input = fake_grid_data.data[rings[j], :]
+        complex_matrix_input = scratch_memory_north[1:nlon÷2 + 1, :, j]
+        real_vector_input = fake_grid_data.data[rings[j], 1]
+        complex_vector_input = scratch_memory_north[1:nlon÷2 + 1, 1, j]
+
+        rfft_plans[j] = FFT_package.plan_rfft(real_matrix_input, 1)
+        brfft_plans[j] = FFT_package.plan_brfft(complex_matrix_input, nlon, 1)
+        rfft_plans_1D[j] = FFT_package.plan_rfft(real_vector_input, 1)
+        brfft_plans_1D[j] = FFT_package.plan_brfft(complex_vector_input, nlon, 1)
+    end
+
+    return rfft_plans, brfft_plans, rfft_plans_1D, brfft_plans_1D
+end
+
+
+function SpeedyTransforms._fourier_batched!(                     # GRID TO SPECTRAL
+    f_north::CuArray{<:Complex, 3},             # Fourier-transformed output
+    f_south::CuArray{<:Complex, 3},             # and for southern latitudes
+    grids::AbstractGridArray{NF, N, <:CuArray}, # gridded input
+    S::SpectralTransform,                       # precomputed transform
+) where {NF<:AbstractFloat, N}
+    (; nlat, nlons, nlat_half) = S          # dimensions
+    (; rfft_plans) = S                      # pre-planned transforms
+    nlayers = size(grids, 2)
+
+    @boundscheck SpeedyTransforms.ismatching(S, grids) || throw(DimensionMismatch(S, grids))
+    @boundscheck nlayers == S.nlayers || throw(DimensionMismatch(S, grids))
+    @boundscheck size(f_north) == size(f_south) == (S.nfreq_max, S.nlayers, nlat_half) || throw(DimensionMismatch(S, grids))
+
+    rings = eachring(grids)                 # precomputed ring indices
+    @inbounds for j_north in 1:nlat_half    # symmetry: loop over northern latitudes only
+        j = j_north                         # symmetric index / ring-away from pole index
+        j_south = nlat - j_north + 1        # corresponding southern latitude index
+        nlon = nlons[j]                     # number of longitudes on this ring
+        nfreq = nlon÷2 + 1                  # linear max Fourier frequency wrt to nlon
+        not_equator = j_north != j_south    # is the latitude ring not on equator?
+
+        rfft_plan = rfft_plans[j]           # FFT planned wrt nlon on ring
+        ilons = rings[j_north]              # in-ring indices northern ring
+        
+        # FOURIER TRANSFORM in zonal direction, northern latitude
+        view(f_north, 1:nfreq, 1:nlayers, j) .= rfft_plan * grids.data[ilons, :]
```

Comment:
@jackleland because at the moment it sounds we would just need an `rfft!`/`brfft!` method that calls this line, either written with `LinearAlgebra.mul!` for CPUs or with something else for GPUs, so the `_fourier_batched!` function is actually the same and we wouldn't need any code duplication here, just a new method for `rfft!`, `brfft!`